examples/controlnet/train_controlnet_flux.py

```diff
@@ -0,0 +1,1321 @@
+                pixel_latents = _pack_latents(
+                    pixel_latents_tmp,
+                    pixel_values.shape[0],
+                    pixel_latents_tmp.shape[1],
+                    pixel_latents_tmp.shape[2],
+                    pixel_latents_tmp.shape[3],
+                )
+
+                control_values = batch["conditioning_pixel_values"].to(dtype=weight_dtype)
+                control_latents = vae.encode(control_values).latent_dist.sample()
+                control_latents = (control_latents - vae.config.shift_factor) * vae.config.scaling_factor
+                control_image = _pack_latents(
+                    control_latents,
+                    control_values.shape[0],
+                    control_latents.shape[1],
+                    control_latents.shape[2],
+                    control_latents.shape[3],
+                )
+
+                latent_image_ids = _prepare_latent_image_ids(
+                    batch_size=pixel_latents_tmp.shape[0],
+                    height=pixel_latents_tmp.shape[2],
+                    width=pixel_latents_tmp.shape[3],
+                    device=pixel_values.device,
+                    dtype=pixel_values.dtype,
+                )
+
+                # Sample noise that we'll add to the latents
+                noise = torch.randn_like(pixel_latents).to(accelerator.device).to(dtype=weight_dtype)
+                bsz = pixel_latents.shape[0]
+
+                # Sample a random timestep for each image
+                t = torch.sigmoid(torch.randn((bsz,), device=accelerator.device, dtype=weight_dtype))
+
+                # apply flow matching
+                noisy_latents = (
+                    1 - t.unsqueeze(1).unsqueeze(2).repeat(1, pixel_latents.shape[1], pixel_latents.shape[2])
+                ) * pixel_latents + t.unsqueeze(1).unsqueeze(2).repeat(
+                    1, pixel_latents.shape[1], pixel_latents.shape[2]
+                ) * noise
+
+                guidance_vec = torch.full(
+                    (noisy_latents.shape[0],), 3.5, device=noisy_latents.device, dtype=weight_dtype
+                )
+
+                controlnet_block_samples, controlnet_single_block_samples = flux_controlnet(
+                    hidden_states=noisy_latents,
+                    controlnet_cond=control_image,
+                    timestep=t,
+                    guidance=guidance_vec,
+                    pooled_projections=batch["unet_added_conditions"]["pooled_prompt_embeds"].to(dtype=weight_dtype),
+                    encoder_hidden_states=batch["prompt_ids"].to(dtype=weight_dtype),
+                    txt_ids=batch["unet_added_conditions"]["time_ids"][0].to(dtype=weight_dtype),
+                    img_ids=latent_image_ids[0],
+                    return_dict=False,
+                )
+
+                noise_pred = flux_transformer(
+                    hidden_states=noisy_latents,
+                    timestep=t,
```

Comment:
Scaling here as well?

Reply:
t was sampled through 
https://github.com/huggingface/diffusers/blob/916fd80a43cba71eca7e309d4310b544144a0107/examples/controlnet/train_controlnet_flux.py#L1190
and it located between 0,1 and no need to scale.